### ISSUE TITLE
Mocha error: Cannot read property 'finally' of undefined

### ISSUE THREAD

**Issue by nolanlawson** [webdriverio/webdriverio#7405]
**Environment:**
 - **WebdriverIO version:** 7.12.3
 - **Mode:** WDIO
 - **If WDIO Testrunner, running sync/async:** Not sure
 - **Node.js version:** 16.5.0
 - **NPM version:** 7.19.1
 - **Browser name and version:** 93.0.4577.63
 - **Platform name and version:** MacOS Big Sur
 - **Additional wdio packages used (if applicable):** See repro

**Describe the bug**

In Mocha mode, when using a `before()` hook, if the function is non-async (doesn't return a Promise), then an error is thrown.

```
TypeError: Cannot read property 'finally' of undefined
    at Context.executeAsync (/Users/nlawson/workspace/repro-wdio-async/node_modules/@wdio/utils/build/shim.js:293:42)
    at Context.testFrameworkFnWrapper (/Users/nlawson/workspace/repro-wdio-async/node_modules/@wdio/utils/build/test-framework/testFnWrapper.js:45:32)
```

This seems to be a recent regression in v7.12.3 of `@wdio/mocha-framework`. In v7.12.2, there is no error.

**To Reproduce**

[Minimal repro](https://gist.github.com/nolanlawson/0020162dc6f248ad03b53b3ad3eeda5e)

To reproduce:

```sh
git clone https://gist.github.com/0020162dc6f248ad03b53b3ad3eeda5e.git repro
cd repro
npm i
npx wdio run ./wdio.conf.js
```

**Expected behavior**

An error should not be thrown for a non-async function.

**Workaround**

Changing the function to `async` (or having it return a `Promise`) resolves the issue.

```patch
 describe('My application', () => {
 
-    before(() => {
+    before(async () => {
         console.log('before')
     })
```

**Comment by dreuxl** [comment]
same here :(

**Comment by christian-bromann** [comment]
Fixed and released as `v7.12.4`

**Comment by dreuxl** [comment]
thanks @christian-bromann
